### ISSUE TITLE
Red Baron (1990) install program: different behavior between MOUNT and IMGMOUNT

### ISSUE THREAD

**Issue by GiulianoPastorino** [joncampbell123/dosbox-x#2194]
Hello,

The game _Red Baron (1990)_ comes with 2 floppy disks for installation.

When trying different methods to install the game, I noticed a discrepancy between the behavior of MOUNT and IMGMOUNT.

**Method 1 — Using MOUNT**

I copied the contents of both disks to the hard drive (to a single folder I named FD), and then added the following lines to [autoexec]:

```
[autoexec]
mount A FD -t floppy
mount C HD
```

I then executed, inside DOSBox-X:

```
C:\> A: [enter]
A:\> INSTALL.COM [enter]
```

A menu with 5 options appears:

![image](https://user-images.githubusercontent.com/77548445/104820419-49c1e980-5813-11eb-8c94-3b5e069cf694.png)

As you can see, option no. 3 allows me to install Red Baron on the hard drive.

**Method 2 — Using IMGMOUNT**

This time, I created images (*.ima) of the diskettes, and then added the following lines to [autoexec]:

```
[autoexec]
imgmount A FD1.ima FD2.ima -t floppy
mount C HD
```

This time, when I executed INSTALL.COM again, the menu that appeared was:

![image](https://user-images.githubusercontent.com/77548445/104820501-c3f26e00-5813-11eb-8442-76e9ae21cc9b.png)

As you can see, the install program didn't recognize a hard drive as in method 1.

**Question:**

Why would MOUNT and IMGMOUNT produce different results from an installation program?

Thank you.

**Comment by rderooy** [comment]
@GiulianoPastorino in the second instance, did you mount a C: drive? If you only mount the floppy disks on A: you will not have an emulated C: drive to install to.

**Comment by GiulianoPastorino** [comment]
@rderooy, yes I did:

```
[autoexec]
imgmount A FD1.ima FD2.ima -t floppy
mount C HD  <———
```

**Comment by rderooy** [comment]
Did you see this thread? It matches your issue exactly.
https://www.vogons.org/viewtopic.php?f=33&t=55895&p=612013&hilit=red%20baron#p612013

In any case, it is not specific to dosbox-x.

Are the disk images made by you from original disks? Do you know if the disks work on a retro PC?

Perhaps the installer is simply buggy

**Comment by GiulianoPastorino** [comment]
> Did you see this thread? It matches your issue exactly.
> https://www.vogons.org/viewtopic.php?f=33&t=55895&p=612013&hilit=red%20baron#p612013

Yes! I initiated that thread :)
I am playing the game, since copying the files manually from the disks works nicely.
My curiosity is why the installation program provides different results when I use MOUNT and IMGMOUNT inside DOSBox or DOSBox-X.

> In any case, it is not specific to dosbox-x.

Indeed. The problem also occurs in DOSBox. I was hoping that DOSBox-X, being a more precise simulation, would provide a more accurate result.

> Are the disk images made by you from original disks? Do you know if the disks work on a retro PC?

Yes, I created the images from the original floppy disks. And yes, when I install Red Baron on a 486DX/100 with a real floppy drive, the installation program does ask for disk number 2. (I mentioned "real" floppy drive because on my current computer I am using a USB floppy drive to read the disks, which I asume comes with its own firmware).

**Comment by Wengier** [comment]
@GiulianoPastorino I believe the install program will check for the volume label of the drive. Please check if the volume label is something like "RBDISK 1". If so, can you change it to e.g. "RBDISK1" (without space) and see the result? You can do so using built-in LABEL command, e.g.

```
LABEL RBDISK1
```

Hope this helps.

**Comment by GiulianoPastorino** [comment]
@Wengier, thank you! That was it. I used IMGMOUNT to mount the floppy images, and the images were without labels. I then labeled them to RBDISK1 and RBDISK2 as you suggested, and it worked perfectly: (1) the installation program now showed the correct menu when using IMGMOUNT; and (2) the installation program asked for disk 2 after copying all the files from disk 1.

Cheers!

**Comment by GiulianoPastorino** [comment]
By the way, quoting myself:

> I was hoping that DOSBox-X, being a more precise simulation, would provide a more accurate result.

I assumed correctly. @Wengier's solution only applies do DOSBox-X, since DOSBox 0.74 does not read the labels from images.

**Comment by GiulianoPastorino** [comment]
Hello everyone!

The solution @Wengier cordially gave me 3 years ago, which was creating labels for Red Baron's floppy disk images, was working perfectly for DOSBox-X version 2022.09.0.

I've decided to change my computer to a new one, and so I've begun reinstalling everything, including DOSBox-X in its newest version: 2023.10.06.

When I was reinstalling Red Baron, the problem I reported in the very beginning of this thread occurred again. The floppy images are the same. The floppy labels are in place. The dosbox-x.conf is the same. The only difference is the DOSBox-X version.

With version 2023.10.06 I get this screen, which lacks the option to install to hard drive:

![image](https://github.com/joncampbell123/dosbox-x/assets/77548445/665aeaeb-71c1-4d62-809a-4a946f058754)

I then removed version 2023.10.06, and installed the old version 2022.09.0, and now I get the desired screen:

![image](https://github.com/joncampbell123/dosbox-x/assets/77548445/c41716d4-2705-4267-8b1f-2d18ea3db2c0)

What could have changed between the versions?

Cheers!

**Comment by maron2000** [comment]
You may want to check similar issues fixed recently.
https://github.com/joncampbell123/dosbox-x/issues?q=is%3Aissue+is%3Aclosed+dynamix

**Comment by GiulianoPastorino** [comment]
Hi, @maron2000,

Thank you for mentioning those issues.

But I believe this particular issue was not fixed, but "unfixed".

The reason is:
- If I install version 2022.09.0, it works.
- If I upgrade to version 2023.10.06, it stops working.

As a curiosity, I also installed an even older version (0.83.9), released in 01/01/21 (before the above mentioned issues), and it also works!!

So something was "unfixed" between 2022.09.0 and 2023.10.06.

What could be it?

Thank you.

**Comment by maron2000** [comment]
Did you really carefully read the issues?
If you still used the hacked floppy image, but the issue is fixed so no more need of hacks.

A quote of your previous comment above.
> I then labeled them to RBDISK1 and RBDISK2 as you suggested, and it worked perfectly:

No hacks, working OK.
![install](https://github.com/joncampbell123/dosbox-x/assets/68574602/420cfd18-c36d-488d-b241-4e4c7a414064)
![title](https://github.com/joncampbell123/dosbox-x/assets/68574602/142acb33-be6a-42ef-a0df-e18b821cc050)

**Comment by GiulianoPastorino** [comment]
Hi, @maron2000,

I did read them. But I didn't find in them any instructions of what to do differently.

(But I only read the issues — I didn't read the pull requests that the issues are mentioning, such as https://github.com/joncampbell123/dosbox-x/pull/3274, for instance.)

> If you still used the hacked floppy image, but the issue is fixed so no more need of hacks.

The original floppy disk carry the following labels: RBDISK 1 and RBDISK 2:

![image](https://github.com/joncampbell123/dosbox-x/assets/77548445/27e4d3f0-b830-48a9-b857-7082a27b6586)

The software I used to create the floppy image, back in 2021, didn't copy the labels. So the images were showing as "has no label":

![image](https://github.com/joncampbell123/dosbox-x/assets/77548445/04c003cd-d4f9-4747-9ab0-f66fa686820b)

@Wengier then suggested me to create the labels: RBDISK1 and RBDISK2. And it worked.

> No hacks, working OK.

Which labels are showing up in your floppy images?

And the curiosity still remains: Why did it work until version 2022.12.26, and stopped working since version 2023.03.31?

**Comment by GiulianoPastorino** [comment]
Hi, @maron2000,

> If you still used the hacked floppy image, but the issue is fixed so no more need of hacks.

Ok, now I understand what you meant by a "hacked" floppy image. I was thinking in the other direction—that is, "why is he calling the action of creating labels on the disks a hack?").

Now I got that the "hack" was creating the label as RBDISK1, not as the original RBDISK 1.

I changed the label from RBDISK1 to RBDISK 1 and the problem was solved.

Thank you.
